cx_Freeze/freezer.py

```diff
@@ -305,7 +305,7 @@ def _GetDependentFiles(self, path, darwinFile: DarwinFile = None) -> List:
         imagehlp library on Windows, otool on Mac OS X and ldd on Linux);
         limit this list by the exclusion lists as needed"""
         path = os.path.normcase(path)
-        dependentFiles = self.dependentFiles.get(path, [])
+        dependentFiles = self.dependentFiles.get(path, None)
```

Comment:
This should be in a separate patch if it is real. I think that with the change when the same dependency is found, the ldd/otool will be run again and detect the same dependencies.

Reply:
Yeah, this one was a separate issues I found when going through the code.  Previously we had "self.dependentFiles.get(path,[])" which would return the list of dependencies, if previously calculated, or [] by default.  The problem was that the code could not distinguish between cases where (i) dependencies had not previously been checked, so that [] was returned by default, and (ii) dependencies were previously checked, but no dependencies were found, so [] as stored in the dictionary.  In case (ii) you could end up running the dependency check code multiple times (unnecessarily) on the same file.  I can put this in a separate PR if you prefer.